formidable/forms/field_builder.py

```diff
@@ -23,10 +23,17 @@ def __init__(self, field):
         self.validator_factory = self.validator_factory_class()
 
     def build(self, role=None):
-        self.access = self.field.accesses.all()[0] if role else None
+        self.access = self.get_accesses(role)
         field_class = self.get_field_class()
         return field_class(**self.get_field_kwargs())
 
+    def get_accesses(self, role):
+        if role:
+            # The role is previously "prefetch" in order to avoid database
+            # hit, we don't use a get() method in queryset.
+            return self.field.accesses.all()[0]
```

Comment:
When using "first" instead of "all()[0]"

```
======================================================================
FAIL: test_queryset_without_role (tests.test_forms.TestDynamicForm)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/moumoutte/projects/django-formidable/demo/tests/test_forms.py", line 305, in test_queryset_without_role
    form.get_django_form_class(role='jedi')
  File "/home/moumoutte/.virtualenvs/formidable/local/lib/python2.7/site-packages/django_perf_rec/api.py", line 91, in __exit__
    self.save_or_assert()
  File "/home/moumoutte/.virtualenvs/formidable/local/lib/python2.7/site-packages/django_perf_rec/api.py", line 127, in save_or_assert
    assert self.record == orig_record, msg
AssertionError: Performance record did not match for TestDynamicForm.test_queryset_without_role
  db: SELECT ... FROM "formidable_field" WHERE "formidable_field"."form_id" = # ORDER BY "formidable_field"."order" ASC
  db: SELECT ... FROM "formidable_item" WHERE "formidable_item"."field_id" IN (...) ORDER BY "formidable_item"."order" ASC
  db: SELECT ... FROM "formidable_access" WHERE ("formidable_access"."access_id" = # AND "formidable_access"."field_id" IN (...))
  db: SELECT ... FROM "formidable_validation" WHERE "formidable_validation"."field_id" IN (...)
  db: SELECT ... FROM "formidable_default" WHERE "formidable_default"."field_id" IN (...)
+ db: SELECT ... FROM "formidable_access" WHERE "formidable_access"."field_id" = # ORDER BY "formidable_access"."id" ASC LIMIT #
+ db: SELECT ... FROM "formidable_access" WHERE "formidable_access"."field_id" = # ORDER BY "formidable_access"."id" ASC LIMIT #
+ db: SELECT ... FROM "formidable_access" WHERE "formidable_access"."field_id" = # ORDER BY "formidable_access"."id" ASC LIMIT #
+ db: SELECT ... FROM "formidable_access" WHERE "formidable_access"."field_id" = # ORDER BY "formidable_access"."id" ASC LIMIT #
  db: SELECT ... FROM "formidable_preset" WHERE "formidable_preset"."form_id" = #
```